cmd/argoexec/commands/wait.go

```diff
@@ -16,7 +14,7 @@ func NewWaitCommand() *cobra.Command {
 		Use:   "wait",
 		Short: "wait for main container to finish and save artifacts",
 		Run: func(cmd *cobra.Command, args []string) {
-			ctx := context.Background()
+			ctx := cmd.Context()
```

Comment:
I suggest putting the code related to signal handling here. The current implementation affects all subcommands of `argoexec`.

Reply:
That’s intentional, as they’re currently not able to deal with SIGTERM. That said, each one needs to be updated to have this line in them.

Reply:
ok~

Reply:
`data` was followed up in https://github.com/argoproj/argo-workflows/pull/12544#discussion_r1522757235